### PR TITLE
Expand JSDoc template generation locations (to 'master')

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
#4962 but in `master`.

This pull request addresses portions of #4447.

We currently now cover
*  methods
* classes
* class constructors
* namespaces
* variables

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -7025,16 +7029,37 @@ namespace ts {
...
+                }
+            }
 
-            if (!containingFunction || containingFunction.getStart() < position) {
+            if (!commentOwner || commentOwner.getStart() < position) {
                 return undefined;
             }
 
-            let parameters = containingFunction.parameters;
+            let parameters = isFunctionLike(commentOwner) ? commentOwner.parameters : emptyArray;
```
Comment (zhengbli) [inline src/services/services.ts]: If the `commentOwner` is a `VariableStatement` that contains a function expression, shouldn't we provide parameters too? E.g.
```
var foo = function (p0, p1, p2) { };
```

Comment (DanielRosenwasser) [inline src/services/services.ts]: Yes, but I think we'll do that in a separate pull request.
